src/nvim/os/env.c

```diff
@@ -346,23 +361,54 @@ void init_homedir(void)
   }
 #endif
 
+  if (var == NULL) {
+    uv_mutex_lock(&mutex);
+    var = os_homedir();
+    uv_mutex_unlock(&mutex);
```

Comment:
looks like the lock/unlock could go in `os_homedir` instead of repeating it at each `os_homedir` call.

Reply:
Done